src/main/java/org/graylog/integrations/aws/resources/AWSResource.java

```diff
@@ -118,4 +124,43 @@ public Response kinesisHealthCheck(@ApiParam(name = "JSON body", required = true
         KinesisHealthCheckResponse response = kinesisService.healthCheck(heathCheckRequest);
         return Response.accepted().entity(response).build();
     }
+
+    /**
+     * Save the new AWS CloudWatch integration.
+     *
+     *  curl 'http://admin:123123123@localhost:9000/api/plugins/org.graylog.integrations/aws/kinesis/save' \
+     *  -v \
+     *  -X POST \
+     *  -H 'X-Requested-By: just-a-test' \
+     *  -H 'Content-Type: application/json' \
+     *  -H 'Accept: application/json' \
+     *  --compressed \
+     *  --data-binary '{
+     *      "aws_access_key": "",
+     *      "aws_secret_key": "",
+     *      "region": "us-east-1",
+     *      "name": "New Flow Logs",
+     *      "description": "Some flow logs.",
+     *      "aws_input_type": "KINESIS_FLOW_LOGS",
+     *      "stream_name": "flow-logs",
+     *      "batch_size": 10000,
+     *      "assume_role_arn": "",
+     *      "global": false,
+     *      "enable_throttling": false
+     * }'
+     */
+    @POST
+    @Timed
+    @Path("/save")
+    @ApiOperation( value = "Save a new Kinesis input" )
```

Comment:
Is this a Kinesis specific resource? If so we should probably change the HTTP path to be more specific. If not, we should probably change the description here.

Reply:
Makes sense. For now, I have changed this to reflect specifically a Kinesis AWS input creation. Each future AWS input type will likely have its own collection of unique required fields. I think separate request value objects should be maintained for each input type (to avoid one massive/confusing common object). 

That being said, there are two types of Kinesis inputs that I want to discuss to sort out a few details with them. I'll message you when I get online tomorrow.